cmd/flow-dps-live/main.go

```diff
@@ -0,0 +1,401 @@
+	if !flagIndexAll && !flagIndexCommit && !flagIndexHeader && !flagIndexPayloads && !flagIndexCollections &&
+		!flagIndexGuarantees && !flagIndexTransactions && !flagIndexResults && !flagIndexEvents && !flagIndexSeals {
+		log.Error().Str("level", flagLevel).Msg("no indexing option specified, use -a/--all to build all indexes")
+		pflag.Usage()
+		return failure
+	}
+
+	// Fail if IndexAll is specified along with other index flags, as this would most likely mean that the user does
+	// not understand what they are doing.
+	if flagIndexAll && (flagIndexCommit || flagIndexHeader || flagIndexPayloads || flagIndexGuarantees ||
+		flagIndexCollections || flagIndexTransactions || flagIndexResults || flagIndexEvents || flagIndexSeals) {
+		log.Error().Str("level", flagLevel).Msg("-a/--all is mutually exclusive with specific indexing flags")
+		pflag.Usage()
+		return failure
+	}
+
+	// Open index database.
+	db, err := badger.Open(dps.DefaultOptions(flagIndex))
+	if err != nil {
+		log.Error().Str("index", flagIndex).Err(err).Msg("could not open index DB")
+		return failure
+	}
+	defer db.Close()
+
+	// Open protocol state database.
+	data, err := badger.Open(dps.DefaultOptions(flagData))
+	if err != nil {
+		log.Error().Err(err).Msg("could not open blockchain database")
+		return failure
+	}
+	defer data.Close()
+
+	// We initialize a metrics logger regardless of whether metrics are enabled;
+	// it will just do nothing if there are no registered metrics.
+	mout := output.New(log, flagMetricsInterval)
+
+	// The storage library is initialized with a codec and provides functions to
+	// interact with a Badger database while encoding and compressing
+	// transparently.
+	var codec dps.Codec
+	codec = zbor.NewCodec()
+	if flagMetrics {
+		size := rcrowley.NewSize("store")
+		mout.Register(size)
+		codec = metrics.NewCodec(codec, size)
+	}
+	storage := storage.New(codec)
+
+	// Check if index already exists.
+	_, err = index.NewReader(db, storage).First()
+	indexExists := err == nil
+	if indexExists && !flagForce {
+		log.Error().Err(err).Msg("index already exists, manually delete it or use (-f, --force) to overwrite it")
+		return failure
+	}
+
+	// The loader component is responsible for loading and decoding the checkpoint.
+	load := loader.New(
+		loader.WithCheckpointPath(flagCheckpoint),
+	)
+
+	nodeID, err := flow.HexStringToIdentifier(flagNodeID)
+	if err != nil {
+		log.Error().Err(err).Msg("invalid node ID")
+		return failure
+	}
+
+	ctx := context.Background()
+	client, err := googlecloud.NewClient(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("could not connect to Google Cloud Platform")
+		return failure
+	}
+
+	bootstrapIdentities, err := network.RetrieveIdentities(flagBootstrap)
+	if err != nil {
+		log.Error().Err(err).Msg("could not retrieve bootstrap identities")
+		return failure
+	}
+
+	blockData := client.Bucket(flagBucket)
+	poller := gcs.NewPoller(blockData)
+
+	execution := execution.New(log, poller, codec, data)
```

Comment:
Sure, can do. That will break all the tests though 😬 

What are your reasons for preferring to have the logic in the poller rather than in the execution follower? I thought the poller might be reused in other places in the future, while the execution follower has only one very specific purpose, so in my mind it made sense to keep the poller as simple and generic as possible